hca/util/__init__.py

```diff
@@ -114,9 +114,13 @@ def __init__(self, retry_after_status_codes={301}, *args, **kwargs):
         self.RETRY_AFTER_STATUS_CODES = frozenset(retry_after_status_codes | retry.Retry.RETRY_AFTER_STATUS_CODES)
 
     def increment(self, *args, **kwargs):
-        retry = super(RetryPolicy, self).increment(*args, **kwargs)
-        logger.warning("Retrying: {}".format(retry.history[-1]))
-        return retry
+        _retry = super(RetryPolicy, self).increment(*args, **kwargs)
+        last_resp = retry.history[-1]
```

Comment:
Can retry.history be empty?

Reply:
not at this point in the code because `super(RetryPolicy, self).increment(*args, **kwargs)` appends the last request to history.